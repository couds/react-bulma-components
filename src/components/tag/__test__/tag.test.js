import React from 'react';
import renderer from 'react-test-renderer';
import Tag from '..';
import CONSTANTS from '../../../constants';

describe('Tag component', () => {
  it('Should exist', () => {
    expect(Tag).toMatchSnapshot();
  });
  it('Should expose Tag Group', () => {
    expect(Tag.Group).toMatchSnapshot();
  });
  it('Should have tag classname', () => {
    const component = renderer.create(<Tag>Tag name</Tag>);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat classname in props with Bulma classname', () => {
    const component = renderer.create(
      <Tag className="other-class this-is-a-test">Tag name</Tag>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use inline styles', () => {
    const component = renderer.create(
      <Tag style={{ width: 250 }}>Tag name</Tag>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be Large', () => {
    const component = renderer.create(<Tag size="large">Tag Large</Tag>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Should be group tags', () => {
    const component = renderer.create(
      <Tag.Group gapless>
        <Tag>Tag 1</Tag>
        <Tag>Tag 2</Tag>
      </Tag.Group>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  [null]
    .concat(Object.keys(CONSTANTS.COLORS).map(key => CONSTANTS.COLORS[key]))
    .map(color =>
      it(`Should use use color ${color}`, () => {
        const component = renderer.create(
          <Tag color={color}>
            tag
            {color}
          </Tag>,
        );
        expect(component.toJSON()).toMatchSnapshot();
      }),
    );
});
