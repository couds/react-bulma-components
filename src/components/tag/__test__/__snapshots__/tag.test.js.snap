// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component Should be Large 1`] = `
<span
  className="tag is-large"
>
  Tag Large
</span>
`;

exports[`Tag component Should be group tags using deprecated gapless 1`] = `
<span
  className="tags has-addons"
>
  <span
    className="tag"
  >
    Tag 1
  </span>
  <span
    className="tag"
  >
    Tag 2
  </span>
</span>
`;

exports[`Tag component Should be group tags using hasAddon (to match bulma docs) 1`] = `
<span
  className="tags has-addons"
>
  <span
    className="tag"
  >
    Tag 1
  </span>
  <span
    className="tag"
  >
    Tag 2
  </span>
</span>
`;

exports[`Tag component Should concat classname in props with Bulma classname 1`] = `
<span
  className="tag other-class this-is-a-test"
>
  Tag name
</span>
`;

exports[`Tag component Should exist 1`] = `[Function]`;

exports[`Tag component Should expose Tag Group 1`] = `[Function]`;

exports[`Tag component Should have tag classname 1`] = `
<span
  className="tag"
>
  Tag name
</span>
`;

exports[`Tag component Should use inline styles 1`] = `
<span
  className="tag"
  style={
    Object {
      "width": 250,
    }
  }
>
  Tag name
</span>
`;

exports[`Tag component Should use use color danger 1`] = `
<span
  className="tag is-danger"
>
  tag
  danger
</span>
`;

exports[`Tag component Should use use color dark 1`] = `
<span
  className="tag is-dark"
>
  tag
  dark
</span>
`;

exports[`Tag component Should use use color info 1`] = `
<span
  className="tag is-info"
>
  tag
  info
</span>
`;

exports[`Tag component Should use use color link 1`] = `
<span
  className="tag is-link"
>
  tag
  link
</span>
`;

exports[`Tag component Should use use color null 1`] = `
<span
  className="tag"
>
  tag
</span>
`;

exports[`Tag component Should use use color primary 1`] = `
<span
  className="tag is-primary"
>
  tag
  primary
</span>
`;

exports[`Tag component Should use use color success 1`] = `
<span
  className="tag is-success"
>
  tag
  success
</span>
`;

exports[`Tag component Should use use color text 1`] = `
<span
  className="tag is-text"
>
  tag
  text
</span>
`;

exports[`Tag component Should use use color warning 1`] = `
<span
  className="tag is-warning"
>
  tag
  warning
</span>
`;
